docs/serialization-guide.md

```diff
@@ -0,0 +1,145 @@
+  * <a name='format-specific-serializers'></a>[Format-specific serializers](serializers.md#format-specific-serializers)
+* <a name='contextual-serialization'></a>[Contextual serialization](serializers.md#contextual-serialization)
+  * <a name='serializers-module'></a>[Serializers module](serializers.md#serializers-module)
+* <a name='deriving-external-serializer-for-another-kotlin-class-experimental'></a>[Deriving external serializer for another Kotlin class (experimental)](serializers.md#deriving-external-serializer-for-another-kotlin-class-experimental)
+  * <a name='external-serialization-uses-properties'></a>[External serialization uses properties](serializers.md#external-serialization-uses-properties)
+<!--- END -->
+
+**Chapter 4.** [Polymorphism](polymorphism.md)
+
+<!--- TOC_REF polymorphism.md -->
+* <a name='closed-polymorphism'></a>[Closed polymorphism](polymorphism.md#closed-polymorphism)
+  * <a name='static-types'></a>[Static types](polymorphism.md#static-types)
+  * <a name='designing-serializable-hierarchy'></a>[Designing serializable hierarchy](polymorphism.md#designing-serializable-hierarchy)
+  * <a name='sealed-classes'></a>[Sealed classes](polymorphism.md#sealed-classes)
+  * <a name='custom-subclass-serial-name'></a>[Custom subclass serial name](polymorphism.md#custom-subclass-serial-name)
+  * <a name='concrete-properties-in-a-base-class'></a>[Concrete properties in a base class](polymorphism.md#concrete-properties-in-a-base-class)
+  * <a name='objects'></a>[Objects](polymorphism.md#objects)
+* <a name='open-polymorphism'></a>[Open polymorphism](polymorphism.md#open-polymorphism)
+  * <a name='registered-subclasses'></a>[Registered subclasses](polymorphism.md#registered-subclasses)
+  * <a name='serializing-interfaces'></a>[Serializing interfaces](polymorphism.md#serializing-interfaces)
+  * <a name='property-of-an-interface-type'></a>[Property of an interface type](polymorphism.md#property-of-an-interface-type)
+  * <a name='static-parent-type-lookup-for-polymorphism'></a>[Static parent type lookup for polymorphism](polymorphism.md#static-parent-type-lookup-for-polymorphism)
+  * <a name='explicitly-marking-polymorphic-class-properties'></a>[Explicitly marking polymorphic class properties](polymorphism.md#explicitly-marking-polymorphic-class-properties)
+  * <a name='registering-multiple-superclasses'></a>[Registering multiple superclasses](polymorphism.md#registering-multiple-superclasses)
+  * <a name='polymorphism-and-generic-classes'></a>[Polymorphism and generic classes](polymorphism.md#polymorphism-and-generic-classes)
+  * <a name='merging-library-serializers-modules'></a>[Merging library serializers modules](polymorphism.md#merging-library-serializers-modules)
+  * <a name='default-polymorphic-type-handler-for-deserialization'></a>[Default polymorphic type handler for deserialization](polymorphism.md#default-polymorphic-type-handler-for-deserialization)
+<!--- END -->
+
+**Chapter 5.** [JSON Features](json.md)
```

Comment:
Currently, everything is structured in the most approachable and consistent way.
Changing the order means careful revisit of all the terminology to avoid forward-references in the guide and probably a lot of restructuring.
This is an extra mile I'm not ready to go in a hurry